src/passes/LegalizeJSInterface.cpp

```diff
@@ -41,38 +41,48 @@ Name SET_TEMP_RET_0("setTempRet0");
 
 struct LegalizeJSInterface : public Pass {
   void run(PassRunner* runner, Module* module) override {
+    // First of all, ensure we have the tempRet global and getter/setter
+    // for it.
+    // With asm2wasm these will almost certainly already exists since
+    // emscripten creates them, but for wasm inputs they always get created
+    // here.
+    addTempRet0Helpers(module);
+
     // for each illegal export, we must export a legalized stub instead
     for (auto& ex : module->exports) {
       if (ex->kind == ExternalKind::Function) {
         // if it's an import, ignore it
-        auto* func = module->getFunction(ex->value);
+        Function* func = module->getFunction(ex->value);
```

Comment:
why not `auto`?

Reply:
Sorry, I'm following llvm and chromium style which says only use auto when the type name appear already on the RHS. e.g. thinks like casts or things that take the type as template arg like make_unique<>.   Otherwise that reader can't infer the type.   I can leave these alone if you prefer.

Reply:
I'd prefer to leave it as `auto` for consistency with the rest of the codebase. But open to discussing changing the style.